path/matchers.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import, unicode_literals
+
+import sys
+import ntpath
+import fnmatch
+
+
+##############################################################################
+# Python 2/3 support
```

Comment:
I'm -1 on having two copies of the same compatibility logic. These should be consolidated into a single module.